web/src/components/upgrade_service/PreflightChecks.tsx

```diff
@@ -43,6 +48,13 @@ const PreflightCheck = ({
 
   useEffect(() => {
     setCurrentStep(1);
+
+    if (
+      !isEqual(prevConfig, config) &&
+      preflightCheck?.preflightResults.length > 0
+    ) {
+      rerunPreflights();
+    }
   }, []);
```

Comment:
wouldn't this rerun preflights whenever the state changes? should there be a separate `useEffect` that only captures `config` and `prevConfig`?

Reply:
@sgalsaleh With an empty dependency, it will only run once, when the page mounts.

Reply:
Ah ok then